### PR TITLE
Dependency cleanup.

### PR DESCRIPTION
**Pull request by DrMarcII**
* Use native.existing_rule to determine whether or not to create
repositories.
* Update versions of most dependencies.
* Add newly added dependency for Selenium.
* Delete doc directory.

### CODE REVIEW

File: web/repositories.bzl [hunk 1]
```diff
@@ -145,6 +146,15 @@ def cglib_nodep():
   )
 
 
+def com_github_blang_semver():
+  go_repository(
+      name="com_github_blang_semver",
+      importpath="github.com/blang/semver",
+      sha256="3d9da53f4c2d3169bfa9b25f2f36f301a37556a47259c870881524c643c69c57",
+      strip_prefix="semver-3.5.1",
+      urls=["https://github.com/blang/semver/archive/v3.5.1.tar.gz"])
```
Comment (jart) [inline web/repositories.bzl]: Add `http://mirror.bazel.build/github.com/blang/semver/archive/v3.5.1.tar.gz` first.

File: web/repositories.bzl [hunk 2]
```diff
@@ -158,19 +168,19 @@ def com_github_gorilla_mux():
   go_repository(
       name="com_github_gorilla_mux",
       importpath="github.com/gorilla/mux",
-      sha256="1a1b35782b0e38534b81d90bb86993fe830a7c1c3974a562554399f850ccdfcd",
-      strip_prefix="mux-1.4.0",
-      urls=["https://github.com/gorilla/mux/archive/v1.4.0.tar.gz"])
+      sha256="e639e6935f3327bed70b583e3311eba262266e79f2c50d436bb99ca70700879f",
+      strip_prefix="mux-1.5.0",
+      urls=["https://github.com/gorilla/mux/archive/v1.5.0.tar.gz"])
```
Comment (jart) [inline web/repositories.bzl]: `http://mirror.bazel.build/github.com/gorilla/mux/archive/v1.5.0.tar.gz`

File: web/repositories.bzl [hunk 3]
```diff
@@ -158,19 +168,19 @@ def com_github_gorilla_mux():
...
   go_repository(
       name="com_github_tebeka_selenium",
       importpath="github.com/tebeka/selenium",
-      sha256="345f204a3ece2469dcc82b59860dec095006e179b1c2ba3e9433b14c90dae167",
-      strip_prefix="selenium-8f4861d1f09c100da29ceec85424c3c96df15170",
+      sha256="22db05f2b3e7b1bfe0f84c45333be798627e1b091918a8c65e241b5b1e9df5cc",
+      strip_prefix="selenium-a7e71178aa40d9b298d464b912c4e831b2be4455",
       urls=[
-          "https://github.com/tebeka/selenium/archive/8f4861d1f09c100da29ceec85424c3c96df15170.tar.gz"
+          "https://github.com/tebeka/selenium/archive/a7e71178aa40d9b298d464b912c4e831b2be4455.tar.gz"
```
Comment (jart) [inline web/repositories.bzl]: `http://mirror.bazel.build/github.com/tebeka/selenium/archive/a7e71178aa40d9b298d464b912c4e831b2be4455.tar.gz`

File: web/repositories.bzl [hunk 4]
```diff
@@ -302,15 +312,30 @@ def junit():
       deps=["@org_hamcrest_core"])
 
 
+def net_bytebuddy():
+  java_import_external(
+      name="net_bytebuddy",
+      jar_sha256=
+      "c7f9861c94b07192ce970820cab9b046dc910e59567855d77c66d4532b9515bf",
+      jar_urls=[
+          "http://central.maven.org/maven2/net/bytebuddy/byte-buddy/1.7.6/byte-buddy-1.7.6.jar",
```
Comment (jart) [inline web/repositories.bzl]: `http://mirror.bazel.build/central.maven.org/maven2/net/bytebuddy/byte-buddy/1.7.6/byte-buddy-1.7.6.jar`

File: web/repositories.bzl [hunk 5]
```diff
@@ -468,10 +493,10 @@ def org_mozilla_firefox():
   platform_http_file(
       name="org_mozilla_firefox",
       amd64_sha256=
-      "6d338e98b1ac1078738dcdab11d6a0c855e5087fa0d7d99dbffa521d16a45327",
+      "9adf41f9c8ed525906d270c66ed2cfbd9e2154a3aeccaadbb0bdfb6ce3a3ca73",
       amd64_urls=[
-          "https://ftp.mozilla.org/pub/firefox/releases/53.0.3/linux-x86_64/en-US/firefox-53.0.3.tar.bz2",
-          "http://mirror.bazel.build/ftp.mozilla.org/pub/firefox/releases/53.0.3/linux-x86_64/en-US/firefox-53.0.3.tar.bz2",
+          "https://ftp.mozilla.org/pub/firefox/releases/56.0.1/linux-x86_64/en-US/firefox-56.0.1.tar.bz2",
+          "http://mirror.bazel.build/https://ftp.mozilla.org/pub/firefox/releases/56.0.1/linux-x86_64/en-US/firefox-56.0.1.tar.bz2",
```
Comment (jart) [inline web/repositories.bzl]: mirror.bazel.build goes first. also should be `http://mirror.bazel.build/ftp.mozilla.org/pub/firefox/releases/56.0.1/linux-x86_64/en-US/firefox-56.0.1.tar.bz2`

File: web/repositories.bzl (hunk 1 above)
Comment (DrMarcII) [reply web/repositories.bzl]: Done.

File: web/repositories.bzl (hunk 2 above)
Comment (DrMarcII) [reply web/repositories.bzl]: Done.

File: web/repositories.bzl (hunk 3 above)
Comment (DrMarcII) [reply web/repositories.bzl]: Done

File: web/repositories.bzl (hunk 4 above)
Comment (DrMarcII) [reply web/repositories.bzl]: Done.

File: web/repositories.bzl (hunk 5 above)
Comment (DrMarcII) [reply web/repositories.bzl]: Done.

**Pull request merged**